core/relation/key.go

```diff
@@ -0,0 +1,6 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package relation
+
+type Key string
```

Comment:
Looking at `state.relationKey`, I'm wondering if the `Key` (which could be renamed IMHO), shouldn't be a `[]string` representing the list of endpoints rather the concat of their names alone.
Unless we are planning to change the contents of the key?

Reply:
I kept it as a string to better integrate with the world as it is. Key is what it's called many places already and seemed to be the best for continuity.